microprofile/grpc/core/src/main/java/io/helidon/microprofile/grpc/core/BidirectionalMethodHandlerSupplier.java

```diff
@@ -0,0 +1,174 @@
+/*
+ * Copyright (c) 2019, 2024 Oracle and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.helidon.microprofile.grpc.core;
+
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.lang.reflect.Type;
+import java.util.function.Supplier;
+
+import io.helidon.grpc.core.GrpcHelper;
+import io.helidon.grpc.core.MethodHandler;
+
+import io.grpc.MethodDescriptor;
+import io.grpc.Status;
+import io.grpc.stub.StreamObserver;
+
+/**
+ * A supplier of {@link MethodHandler}s for bi-directional streaming gRPC methods.
+ */
+public class BidirectionalMethodHandlerSupplier extends AbstractMethodHandlerSupplier {
+
+    /**
+     * Create a supplier of handlers for bi-directional streaming methods.
+     */
+    // method is public because it is loaded via ServiceLoader
+    public BidirectionalMethodHandlerSupplier() {
```

Comment:
Please do not use public constructors.

Reply:
These are public because of ServiceLoader, as explained in the comment